File: cs/proxy/hall/ShopOperation.cs
/// <summary>
/// 商城操作
/// </summary>
namespace org.zgl.service.client.hall{
	[Proxy(gameId = -10)]
	public interface ShopOperation : ITcpAsyncService { 
		/// <summary>
		/// 购买vip
		/// </summary>
		void pay(int productId);
		/// <summary>
		/// 购买vip ----> <<pay();回调>>
		/// </summary>
		void pay2CallBack(ItemListDto callBackParam);
	}
}

File: componet/src/main/java/org/zgl/service/client/hall/ShopOperation.java
package org.zgl.service.client.hall;

/**
 * @author： 猪哥亮
 * @创建时间： 2018/8/17
 * @文件描述：
 */

import org.zgl.dto.ItemListDto;
import org.zgl.proxy.ClassDesc;
import org.zgl.proxy.MethodDesc;
import org.zgl.proxy.rule.ITcpAsyncService;

/**
 * 商城操作
 */
@ClassDesc("商城操作")
public interface ShopOperation extends ITcpAsyncService {
    @MethodDesc("购买vip")
    ItemListDto pay(int productId);
}
